sbncode/LArRecoProducer/ShowerCosmicCylinder_module.cc

```diff
@@ -0,0 +1,170 @@
+////////////////////////////////////////////////////////////////////////
+// Class:       ShowerCosmicCylinder
+// Plugin Type: producer (art v3_03_01)
+// File:        ShowerCosmicCylinder_module.cc
+//
+// Generated at Tue Sep 15 08:54:39 2020 by Edward Tyley using cetskelgen
+// from cetlib version v3_08_00.
+//
+// Producer to find the distance of closest approach from a shower to
+// a unambiguous cosmic ray
+////////////////////////////////////////////////////////////////////////
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "canvas/Utilities/InputTag.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "lardata/Utilities/AssociationUtil.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include "canvas/Persistency/Common/FindManyP.h"
+#include "lardataobj/RecoBase/PFParticle.h"
+#include "lardataobj/RecoBase/PFParticleMetadata.h"
+#include "lardataobj/RecoBase/Shower.h"
+#include "lardataobj/RecoBase/SpacePoint.h"
+#include "lardataobj/RecoBase/Track.h"
+
+#include <memory>
+
+#include "TF1.h"
+#include "TGraph.h"
+
+namespace sbn {
```

Comment:
What is the point of these little forward declarations? Does `cetskelgen` or something make them?

Reply:
This is a `cetskelgen` feature

Reply:
I love forward declarations in general, but they're definitely pointless when there is no code between them and the full declaration. I would remove them.